File: lib/js/src/components/ShareButtonStyles.js
'use strict';

var Block        = require("bs-platform/lib/js/block.js");
var Colors       = require("../styles/Colors.js");
var Glamor       = require("bs-glamor/lib/js/src/glamor.js");
var ButtonStyles = require("../styles/ButtonStyles.js");

var root = Glamor.css(/* :: */[
      Glamor.position("relative"),
      /* :: */[
        /* Selector */Block.__(1, [
            "& input",
            /* :: */[
              Glamor.background(Colors.panelDark),
              /* :: */[
                Glamor.transition("all 250ms"),
                /* :: */[
                  Glamor.width("0"),
                  /* :: */[
                    Glamor.padding("0"),
                    /* :: */[
                      Glamor.color("#888"),
                      /* [] */0
                    ]
                  ]
                ]
              ]
            ]
          ]),
        /* :: */[
          /* Selector */Block.__(1, [
              "& .tooltip",
              /* :: */[
                Glamor.display("none"),
                /* :: */[
                  Glamor.position("absolute"),
                  /* :: */[
                    Glamor.zIndex("100"),
                    /* :: */[
                      Glamor.top("100%"),
                      /* :: */[
                        Glamor.right("1em"),
                        /* :: */[
                          Glamor.background("rgba(0, 0, 0, .6)"),
                          /* :: */[
                            Glamor.color("#aaa"),
                            /* :: */[
                              Glamor.whiteSpace("nowrap"),
                              /* :: */[
                                Glamor.padding(".4em .8em"),
                                /* :: */[
                                  Glamor.borderRadius(".25em"),
                                  /* :: */[
                                    Glamor.fontSize(".8rem"),
                                    /* :: */[
                                      /* Selector */Block.__(1, [
                                          "& .arrow",
                                          /* :: */[
                                            Glamor.position("absolute"),
                                            /* :: */[
                                              Glamor.content(" "),
                                              /* :: */[
                                                Glamor.bottom("100%"),
                                                /* :: */[
                                                  Glamor.right("2em"),
                                                  /* :: */[
                                                    Glamor.height("0"),
                                                    /* :: */[
                                                      Glamor.width("0"),
                                                      /* :: */[
                                                        Glamor.border(".5em solid transparent"),
                                                        /* :: */[
                                                          Glamor.borderBottomColor("rgba(0, 0, 0, .6)"),
                                                          /* :: */[
                                                            Glamor.marginLeft(".5em"),
                                                            /* [] */0
                                                          ]
                                                        ]
                                                      ]
                                                    ]
                                                  ]
                                                ]
                                              ]
                                            ]
                                          ]
                                        ]),
                                      /* :: */[
                                        /* Selector */Block.__(1, [
                                            "& .confirmation-message",
                                            /* :: */[
                                              Glamor.display("none"),
                                              /* :: */[
                                                Glamor.padding("0 .75em"),
                                                /* [] */0
                                              ]
                                            ]
                                          ]),
                                        /* [] */0
                                      ]
                                    ]
                                  ]
                                ]
                              ]
                            ]
                          ]
                        ]
                      ]
                    ]
                  ]
                ]
              ]
            ]),
          /* :: */[
            /* Selector */Block.__(1, [
                "&:hover input",
                /* :: */[
                  Glamor.width("25vw"),
                  /* :: */[
                    Glamor.marginRight("1em"),
                    /* :: */[
                      Glamor.padding(".3em"),
                      /* [] */0
                    ]
                  ]
                ]
              ]),
            /* :: */[
              /* Selector */Block.__(1, [
                  "&:hover .tooltip, &.s-show-confirmation .tooltip",
                  /* :: */[
                    Glamor.display("block"),
                    /* [] */0
                  ]
                ]),
              /* :: */[
                /* Selector */Block.__(1, [
                    "&.s-show-confirmation",
                    /* :: */[
                      /* Selector */Block.__(1, [
                          "& .tooltip .message",
                          /* :: */[
                            Glamor.display("none"),
                            /* [] */0
                          ]
                        ]),
                      /* :: */[
                        /* Selector */Block.__(1, [
                            "& .tooltip .confirmation-message",
                            /* :: */[
                              Glamor.display("block"),
                              /* [] */0
                            ]
                          ]),
                        /* [] */0
                      ]
                    ]
                  ]),
                /* [] */0
              ]
            ]
          ]
        ]
      ]
    ]);

var button = ButtonStyles.button;

exports.root   = root;
exports.button = button;
/* root Not a pure module */

File: src/common/components/ShareButtonStyles.ml
open TypedGlamor

let container ~showConfirmation = css [
  position relative;

  select "& input" [
    background Colors.panelDark;
    transitions [AnimatableProperty.all, (ms 250), easeInOut, (ms 0)];
    width zero;
    padding zero;
    color (rgba 255 255 255 0.75);
  ];

  select "& .tooltip" [
    position absolute;
    display none;
    zIndex (int 100);
    offsetTop (pct 100.);
    offsetRight (em 1.);
    background (rgba 0 0 0 1.);
    color (hex "ccc");
    whiteSpace nowrap;
    padding2 ~v:(em 0.4) ~h:(em 0.8);
    borderRadius (em 0.25);
    fontSize (rem 0.8);

    select "::before" [
      position absolute;
      unsafe "content" " ";
      offsetBottom (pct 100.);
      offsetRight (em 2.);
      height zero;
      width zero;
      border3 (em 0.5) solid transparent;
      (*pointerEvents "none";*)
      borderBottomColor (rgba 0 0 0 1.);
      marginLeft (em 0.5);
    ];

    select "& .confirmation-message" [
      display none;
      padding2 ~v:zero ~h:(em 0.75);
    ];

    if showConfirmation then
      add [
        display block;

        select "& .confirmation-message" [
          display block;
        ];

        select "& .message" [
          display none;
        ];
      ]
    else
      nothing;
  ];

  hover [
    select "& input" [
      width (vw 25.);
      marginRight (em 1.);
      padding (em 0.3);
    ];

    select "& .tooltip" [
      display block;
    ];
  ];
]